cpp/src/io/statistics/column_stats.cu

```diff
@@ -198,35 +195,21 @@ gatherIntColumnStats(stats_state_s *s, statistics_dtype dtype, uint32_t t, Stora
     s->ck.non_nulls  = nn_cnt;
     s->ck.null_count = s->group.num_rows - nn_cnt;
   }
-  vmin = warp_reduce(storage.integer_stats[t / 32]).Reduce(vmin, cub::Min());
-  vmin = shuffle(vmin);
-  vmax = warp_reduce(storage.integer_stats[t / 32]).Reduce(vmax, cub::Max());
-  vmax = shuffle(vmax);
-  vsum = warp_reduce(storage.integer_stats[t / 32]).Sum(vsum);
-  if (!(t & 0x1f)) {
-    s->warp_min[t >> 5].i_val = vmin;
-    s->warp_max[t >> 5].i_val = vmax;
-    s->warp_sum[t >> 5].i_val = vsum;
-  }
-  has_minmax = __syncthreads_or(vmin <= vmax);
-  if (t < 32 * 1) {
-    vmin = warp_reduce(storage.integer_stats[t / 32]).Reduce(s->warp_min[t].i_val, cub::Min());
-    if (!(t & 0x1f)) {
-      s->ck.min_value.i_val = vmin;
-      s->ck.has_minmax      = (has_minmax);
-    }
-  } else if (t < 32 * 2) {
-    vmax =
-      warp_reduce(storage.integer_stats[t / 32]).Reduce(s->warp_max[t & 0x1f].i_val, cub::Max());
-    if (!(t & 0x1f)) { s->ck.max_value.i_val = vmax; }
-  } else if (t < 32 * 3) {
-    vsum = warp_reduce(storage.integer_stats[t / 32]).Sum(s->warp_sum[t & 0x1f].i_val);
-    if (!(t & 0x1f)) {
-      s->ck.sum.i_val = vsum;
-      // TODO: For now, don't set the sum flag with 64-bit values so we don't have to check for
-      // 64-bit sum overflow
-      s->ck.has_sum = (dtype <= dtype_int32 && has_minmax);
-    }
+  vmin = block_reduce(storage.integer_stats).Reduce(vmin, cub::Min());
+  __syncthreads();
+  vmax = block_reduce(storage.integer_stats).Reduce(vmax, cub::Max());
+  __syncthreads();
+  vsum = block_reduce(storage.integer_stats).Sum(vsum);
+
+  if (!t) {
+    bool has_minmax       = (vmin <= vmax);
+    s->ck.min_value.i_val = vmin;
+    s->ck.max_value.i_val = vmax;
+    s->ck.has_minmax      = has_minmax;
+    s->ck.sum.i_val       = vsum;
+    // TODO: For now, don't set the sum flag with 64-bit values so we don't have to check for
+    // 64-bit sum overflow
+    s->ck.has_sum = (dtype <= dtype_int32 && has_minmax);
```

Comment:
Very minor - if the chunks are initialized, we should be able to skip this whole block if `has_minmax` is false.